gravityflow/gravityflowformconnector
# Javascript Error When Processing Key Field Choices for Field Mapping

In Gravityforms 2.5, a generic map field processes the choices for a given key field by using `Array.prototype.map()` on the array of choices. 

However, the value being passed to the React component had defined numeric keys ("0", "1", "2", etc) because the `array_filter()` call stripping out the current field causes the keys not to be sequential. Whenever this occurs, Javascript interprets it as an associative array, and encodes it as an `Object` in javascript. 

This fix simply wraps the returned array in `array_values`, which essentially resets the key ordering to be totally sequential, thus solving the issue. 

Addresses issue 117: https://github.com/gravityflow/backlog/issues/117

## Testing Instructions
1. Create a new form and add some fields. 
2. Add a Workflow step to "Create New Entry"
3. When the page reloads, choose a specific form to use for mapping. 
4. Ensure you see the UI update correctly, there are no JS errors in the console, and that you can complete the flow as intended.
